### ISSUE TITLE
Allow matcher to specify user address that can withdraw earned fees

### ISSUE THREAD

**Issue by partylikeits1983** [compolabs/orderbook-contract#60]
Currently, the order book smart contract adds the earned fee of the matching transaction to the `msg_sender()`: 
https://github.com/compolabs/orderbook-contract/blob/465abfed3a0e427774e1a871e6a60b003c6700c0/market-contract/src/main.sw#L204
https://github.com/compolabs/orderbook-contract/blob/465abfed3a0e427774e1a871e6a60b003c6700c0/market-contract/src/main.sw#L207

However, since on the spark matcher we store the private key in a `.env` file, anyone who has access to the server that is running the spark matcher, will have access to the private key that can withdraw earned fees. 

Private key in matcher is stored in a `.env` file: https://github.com/compolabs/spark-matcher-rs/blob/bc646eddf68faab51cb943ef348c9e2697f143ae/.env.example#L3

It might be best to allow the user calling the `match_order_many()` function to specify the address they want to receive the fee. This way, the private key that is stored on the server running the spark matcher only has access to a minimal amount of assets if access to the server is compromised. 

I was thinking something like this (pseudo code) :

```rust
#[storage(read, write)]
fn match_order_internal(
    order0_id: b256,
    order0: Order,
    order1_id: b256,
    order1: Order,
    fee_receiver: Option<Identity>, // Added fee_receiver as an Option 
) -> (MatchResult, b256, u64) {
    // Determine the fee receiver address
    let matcher = match fee_receiver {
        Some(Identity) => Identity,
        None => msg_sender().unwrap(),
    };

    // existing logic...
}
```

TL;DR: why add this? Because if the private key on the matching server is compromised, the earned fees from the matcher are not.

**Comment by partylikeits1983** [comment]
Other ideas of resolving this are:

1) Modify spark-matcher rust code to send earned fees to a specified address once accumulated fees reach a certain threshold 
2) Modify order book contract logic to send fees to address that the order matcher specifies
3) Keep accumulated fees in order book smart contract, that are withdrawn by specified address (same as 2, but without `transfer()`)

**Comment by partylikeits1983** [comment]
To minimize the complexity of the contracts, the issues brought up in this issue need to be handled by the matcher, not the contracts.
